### DISCUSSION TITLE
C# 10 namespace and partial class files

### DISCUSSION THREAD

**Discussion by tomschrot** [dotnet/csharplang#5275] [Q&A]
I just discovered that writing a class with multiple source files as 

`public static partial class xxxx {}`

do only require one single namespace  xyz; - statement in just one of the partial source files!?

What is the recommended, correct way to write partial classes?

**Comment by CyrusNajmabadi** [comment]
> do only require one single namespace xyz; - statement in just one of the partial source files!?

No.  You will need the namespace in all of htem.

when you have:

File1.cs
```c#
namespace Foo;

public partial class C { }
```

This is defining a class `Foo.C`.

When you have:

File2.cs
```c#
public partial class C { }
```

It's just defining a class `C` in teh global namespace.  As these are two different types, they are not combined at all into one single type.

**Comment by CyrusNajmabadi** [comment]
> What is the recommended, correct way to write partial classes?

There is no recommended way.  However, if you are going use them, then it needs to follow the rules of the language (for example, that all parts must be `partial`). Similarly, you can decide if you want to include things like base-types, or generic-constraints, on one, or some, or all of the parts.

**Comment by tomschrot** [comment]
Thank you!
